compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -620,15 +629,14 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         (args, arg_count)
     }
 
-    #[instrument(level = "debug", skip_all)]
+    #[instrument(level = "debug", skip(self))]
     pub fn lower_generic_args_of_assoc_item(
         &self,
         span: Span,
         item_def_id: DefId,
         item_segment: &hir::PathSegment<'tcx>,
         parent_args: GenericArgsRef<'tcx>,
     ) -> GenericArgsRef<'tcx> {
-        debug!(?span, ?item_def_id, ?item_segment);
```

Comment:
Note that I wrote tracing code like this intentionally in HIR ty lowering and other places (i.e., `skip_all` and tracing immediately). Semantically they do very different things: 

Instrumented parameters don't get traced immediately or **at all**, it all depends on the existence of tracing calls. They're simply added to the tracing span.

For example, `lower_generic_args_of_assoc_item` would've never traced all its parameters if it wasn't for the `debug!(?generics);` in `lower_generic_args_of_path`! That's super fragile!

I'm pretty sure some of my tracing instrumentations have been rewritten already. Sadly, it's hard to communicate such subtlety ^^'. Maybe there's a way to force the `tracing::instrument` / rustc's subscriber to trace on function entry, I'm not an expert.